### PR TITLE
Create Bets

### PR DESCRIPTION
**Pull request by ghost**
Dependent on #28
Allows logged in users to create bets by specifying their text and expiry date.
Created bets can be viewed separately.
All bets will be available through /bets page, where user will be able to specify search criteria to filter through the database.
Ongoing (active bets with status set to 1) will be displayed on the main page

### CODE REVIEW

File: app/views/bets/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      </div>
+
+      <div class="form-group">
+        <button type="submit" class="btn btn-primary">Create bet</button>
+      </div>
+    </form>
+
+  <% else %>
+    <h2>You are not logged in </h2>
+<% end %>
```
Comment (jacquesporveau) [inline app/views/bets/new.html.erb]: Identation on the `end`

Also: curious why we aren't using a Rails form_for tag?

File: app/views/layouts/application.html.erb [hunk 2]
```diff
@@ -9,6 +9,10 @@
   </head>
 
   <body>
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery-datetimepicker/2.5.4/build/jquery.datetimepicker.full.min.js"></script>
+
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/jquery-datetimepicker/2.5.4/build/jquery.datetimepicker.min.css" />
+    <link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" rel="stylesheet">
```
Comment (jacquesporveau) [inline app/views/layouts/application.html.erb]: Looks like Rails is loading the other js and css in the head tag. Putting these in the body seems a bit odd.

File: app/views/pages/index.html.erb [hunk 3]
```diff
@@ -1,14 +1,55 @@
...
   <% if current_user %>
-    <h2>Logged-in User:</h2>
+    <div class="row">
+      <div class="col col-lg-4">
+        <%= link_to 'Create Bet', new_bet_path %>
+      </div>
+      <div class="col col-lg-4">
+      </div>
+      <div class="col col-lg-3">
+      </div>
```
Comment (jacquesporveau) [inline app/views/pages/index.html.erb]: Up to you for style preferences but I think we could single line these last couple lines
`<div class="col col-lg-3></div>` etc.

File: app/views/pages/index.html.erb [hunk 4]
```diff
@@ -1,14 +1,55 @@
...
+      <div class="col col-lg-1">
+        <div class="dropdown">
+            <img class="" type="button" data-toggle="dropdown" src="http://fuuse.net/wp-content/uploads/2016/02/avatar-placeholder.png" style="width: 100%;">
```
Comment (jacquesporveau) [inline app/views/pages/index.html.erb]: 4 space indent and not sure we aren't using the Rails image_tag?
http://apidock.com/rails/ActionView/Helpers/AssetTagHelper/image_tag
Also noticing you have a class attr with no data. Not sure if needed?

File: app/views/pages/index.html.erb [hunk 5]
```diff
@@ -1,14 +1,55 @@
...
+        <h2>Ongoing bets</h2>
+      </div>
+    </div>
+    <div class="row">
+      <table class="table table-hover">
+        <thead class="thead-default">
+          <tr>
+            <th class="col-md-2">
+              Created By
+            </th>
```
Comment (jacquesporveau) [inline app/views/pages/index.html.erb]: You can one line these all if you want. Up to you but I think it looks better.

File: app/views/pages/index.html.erb [hunk 6]
```diff
@@ -1,14 +1,55 @@
...
+      </table>
+    </div>
   <% else %>
-    <h2>Welcome, stranger!</h2><br>
+    <div class="row text-center">
+      <div class="col col-lg-4 offset-lg-4">
+        <h2>Welcome, stranger!</h2><br>
+      </div>
+    </div>
       <div class="row">
```
Comment (jacquesporveau) [inline app/views/pages/index.html.erb]: Not part of the PR but your identation is off here too.

**Review by jacquesporveau — COMMENTED** [review]
Looking good! Left some general feedback. Most of it is optional.

File: app/views/layouts/application.html.erb (hunk 2 above)
Comment (ghost) [reply app/views/layouts/application.html.erb]: Thank you very much for all the feedback, I'll get to fixing it!
Yeah, I actually tried putting it in the head tag but it wouldn't load for some reason. Maybe I did something wrong, I'll try again.

File: app/views/bets/new.html.erb (hunk 1 above)
Comment (ghost) [reply app/views/bets/new.html.erb]: Good catch, thanks! 
Just feels like doing it manually gives me a bit more control over the style, but I might be mistaken here. Is it considered a better idea to use form_for in situations like that?

Comment (jacquesporveau) [reply app/views/bets/new.html.erb]: It's considered to be best practices. It's kind of like buying a state of the art chainsaw and then using a handsaw instead.
I think some of the philosophy behind Rails is "lets make things easy". By using the form for tag you are using the Rails magic.

File: app/views/bets/new.html.erb [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<div class="container">
+  <% if current_user %>
+    <h2> Creating a new bet </h2>
+    <form action="<% create_bet_path %>" method="post" id="create-bet-form">
```
Comment (isaacfreeman) [inline app/views/bets/new.html.erb]: In situations like this where there's no instance to to use with `form_for`, Rails provides `form_tag` and corresponding methods like `text_field_tag`, `hidden_field_tag` etc.

File: app/views/pages/index.html.erb [hunk 8]
```diff
@@ -1,21 +1,52 @@
...
+<div class="container">
   <% if current_user %>
-    <h2>Logged-in User:</h2>
+    <div class="row">
+      <div class="col col-lg-4">
+        <%= link_to 'Create Bet', new_bet_path %>
+      </div>
+      <div class="col col-lg-4"></div>
+      <div class="col col-lg-3"></div>
+      <div class="col col-lg-1">
```
Comment (isaacfreeman) [inline app/views/pages/index.html.erb]: Can you use a Bootstrap offset class here to replace the empty tags above?

File: app/views/pages/index.html.erb [hunk 9]
```diff
@@ -1,21 +1,52 @@
...
+        <div class="dropdown">
+          <img type="button" data-toggle="dropdown" src="http://fuuse.net/wp-content/uploads/2016/02/avatar-placeholder.png" style="width: 100%;">
```
Comment (isaacfreeman) [inline app/views/pages/index.html.erb]: It looks like this image represents a page control. It's probably best to use a `<button>` here with the image specified in the CSS, and keep `` for images that are content rather than controls.

File: config/routes.rb [hunk 10]
```diff
@@ -8,5 +8,7 @@
 
   get '/signup', to: 'users#new', as: :new_user_path
 
+  get  '/create_bet', to: 'bets#new', as: :new_bet
```
Comment (qr8r) [inline config/routes.rb]: You should be using the `resources` syntax in order to create RESTful routes. 

http://guides.rubyonrails.org/routing.html

File: app/controllers/bets_controller.rb [hunk 11]
```diff
@@ -1,5 +1,22 @@
 class BetsController < ApplicationController
-  def new_bet
+  def new
```
Comment (qr8r) [inline app/controllers/bets_controller.rb]: Should be squashed

File: app/views/pages/index.html.erb [hunk 12]
```diff
@@ -1,21 +1,52 @@
...
+      </div>
+    </div>
     <hr>
-    <%= current_user.username %><br>
-    <%= current_user.email %>
-    <%= link_to 'log out', session_path, method: :delete %>
+
+    <div class="row">
+      <div class="col col-md-4">
+        <h2>Ongoing bets</h2>
```
Comment (qr8r) [inline app/views/pages/index.html.erb]: This content would be better suited for a bet/index page now

File: app/views/pages/index.html.erb [hunk 13]
```diff
@@ -1,21 +1,52 @@
...
+      <div class="col col-lg-4">
+        <%= link_to 'Create Bet', new_bet_path %>
+      </div>
+      <div class="col col-lg-4"></div>
+      <div class="col col-lg-3"></div>
+      <div class="col col-lg-1">
+        <div class="dropdown">
+          <img type="button" data-toggle="dropdown" src="http://fuuse.net/wp-content/uploads/2016/02/avatar-placeholder.png" style="width: 100%;">
+          <ul class="dropdown-menu">
+            <li><%= link_to 'Log Out', session_path, method: :delete %></li>
```
Comment (qr8r) [inline app/views/pages/index.html.erb]: Since this is content that is likely shared across many pages, It should be extracted into a partial and rendered in the application layout

File: app/controllers/bets_controller.rb [hunk 14]
```diff
@@ -1,5 +1,22 @@
...
   end
+
+  def create
+    @bet = Bet.new(bet_params)
+    if @bet.save
+      flash[:success] = 'Bet has been created'
+      redirect_to root_path
+    else
+      flash[:danger] = 'Could not create bet'
+      redirect_to create_bet_path
```
Comment (qr8r) [inline app/controllers/bets_controller.rb]: Should render the create page so that the form fields stay populated with the values the user tried to use.

File: app/controllers/bets_controller.rb [hunk 15]
```diff
@@ -1,5 +1,22 @@
...
-  def new_bet
+  def new
     @bet = Bet.new
   end
+
+  def create
+    @bet = Bet.new(bet_params)
+    if @bet.save
+      flash[:success] = 'Bet has been created'
+      redirect_to root_path
```
Comment (qr8r) [inline app/controllers/bets_controller.rb]: Should redirect to the Bet#show page
Same with update when you get there. 

After you create something, as a user, dont you want to see the thing you created?

File: app/views/bets/new.html.erb [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+      <div>
+        <input type="hidden" value="<%= form_authenticity_token %>" name="authenticity_token"/>
```
Comment (qr8r) [inline app/views/bets/new.html.erb]: You might want to stick to the rails form helpers. They provide a lot of begin the scenes help for you. 

Once example, is that if you try and edit an existing record. All of the records fields will be pre populated in the form fields
http://guides.rubyonrails.org/form_helpers.html

File: config/routes.rb [hunk 17]
```diff
@@ -9,6 +9,7 @@
   get '/signup', to: 'users#new', as: :new_user_path
 
   get  '/create_bet', to: 'bets#new', as: :new_bet
+  post '/create_bet', to: 'bets#create', as: :create_bet
```
Comment (qr8r) [inline config/routes.rb]: use the rails `:resources` syntax. Explicitly defining routes with 'get' etc should be exceptional and used only when absolutely needed

**Review by qr8r — CHANGES_REQUESTED** [review]
I would like this PR to include a Bet index page and a Bet show page.

They dont need to be fancy or styled. Just present and functional

**Review by isaacfreeman — APPROVED** [review]

File: app/views/pages/index.html.erb (hunk 12 above)
Comment (ghost) [reply app/views/pages/index.html.erb]: My idea was to have all the current ongoing (active) bets (sorted by date) listed right on the front page and all the bets ever created on bet/index (with the ability to search/filter/etc.). Would it make sense to leave it here for now in this case?

**Comment by YarekTenko** [comment]
Thanks for the feedback, everybody, hope I fixed most of the problems this time.

**Review by qr8r — CHANGES_REQUESTED** [review]
This PR breaks the site

File: app/assets/stylesheets/application.scss [hunk 18]
```diff
@@ -20,3 +20,4 @@
 @import "bootstrap";
 @import "font-awesome";
 @import "session-access";
+@import "user-bar";
```
Comment (qr8r) [inline app/assets/stylesheets/application.scss:23]: This commit needs a commit message

Adding a controller action and styling page elements should be two separate commits. 
Remember a commit is the smallest piece of coherent and functional code you can put together.

Adding a controller action and Adding a top navigation bar are unrelated and should not be in the same PR, let alone the same commit

File: app/views/pages/index.html.erb [hunk 19]
```diff
@@ -1,15 +1,20 @@
...
-    <%= link_to 'Log Out', session_path, method: :delete %>
-    <%= link_to 'Edit Profile', edit_user_path(current_user) %>
-  <% else %>
-    <h2>Welcome, stranger!</h2>
+    <div class="row">
+      <div class="col col-md-4">
+        <h2>Ongoing bets</h2>
+      </div>
+    </div>
+    <%= render 'bets/display_bets' %>
```
Comment (qr8r) [inline app/views/pages/index.html.erb]: This is beyond the scope of the story. 

The story asks only for a page you can visit to create a new bet

File: config/routes.rb [hunk 20]
```diff
@@ -8,5 +8,6 @@
 
   get '/signup', to: 'users#new', as: :new_user_path
 
+  resources :bets
```
Comment (qr8r) [inline config/routes.rb:11]: Are all bet routes available to you at this point?

File: app/controllers/sessions_controller.rb [hunk 21]
```diff
@@ -5,7 +5,7 @@ def create
     user = User.find_by(email: params[:session][:email])
     if user && user.authenticate(params[:session][:password])
```
Comment (qr8r) [inline app/controllers/sessions_controller.rb]: ^^ Commit message is empty 

Why are you making this change?

File: app/assets/javascripts/application.js [hunk 22]
```diff
@@ -14,5 +14,6 @@
 //= require tether
```
Comment (qr8r) [inline app/assets/javascripts/application.js:14]: This is a good sized commit

**Review by qr8r — CHANGES_REQUESTED** [review]

File: app/views/layouts/application.html.erb [hunk 23]
```diff
@@ -13,6 +13,7 @@
       <% flash.each do |key, value| %>
         <div class="alert alert-<%= key %>"><%= value %></div>
       <% end %>
+      <%= render partial: 'shared/user_bar' %>
```
Comment (qr8r) [inline app/views/layouts/application.html.erb]: I believe you dont need to specify partial here.

from inside a view file rails automatically assumes it will be a partial.

<%= render shared/user_bar' %>

**Review by qr8r — APPROVED** [review]
Much better PR! 

One tiny comment, but its nto enough to ask for changes here
